Watchman.Discord/Areas/Responses/Services/ResponsesGetterService.cs

```diff
@@ -25,14 +29,39 @@ public IEnumerable<Response> GetResponsesFromBase()
 
         public IEnumerable<Response> GetResponsesFromResources()
         {
+            Log.Information("Initializing default responses...");
+            var managerMethods = typeof(ResponsesManager).GetMethods();
             var defaultResponses = typeof(Devscord.DiscordFramework.Framework.Commands.Responses.Resources.Responses).GetProperties()
                 .Where(x => x.PropertyType.Name == "String")
                 .Select(prop =>
                 {
                     var onEvent = prop.Name;
                     var message = prop.GetValue(prop)?.ToString();
-                    return new Response(onEvent, message, DEFAULT_SERVER_ID);
+                    var parameters = managerMethods.First(info => info.Name == onEvent).GetParameters().Skip(1);
+                    var availableVariables = new List<string>();
+                    foreach (var parameter in parameters)
+                    {
+                        switch (parameter.ParameterType.Name)
+                        {
+                            case "Contexts":
+                                availableVariables.Add("user");
+                                availableVariables.Add("channel");
+                                availableVariables.Add("server");
+                                break;
+                            case "UserContext":
+                                availableVariables.Add("user");
+                                break;
+                            case "DiscordServerContext":
+                                availableVariables.Add("server");
+                                break;
+                            default:
+                                availableVariables.Add(parameter.Name);
+                                break;
+                        }
```

Comment:
co myślisz o tym? Zobacz czy byś mógł coś podobnego do kodu wprowadzić.
``
```suggestion
                        var variables = parameter.ParameterType.Name switch
                        {
                            "Contexts" => new List { "user", "channel", "server" },
                            "UserContext" => new List { "user" },
                            "DiscordServerContext" => new List { "server" },
                            _ => new List { parameter.Name }
                        };
                        availableVariables.AddRange(variables);

Reply:
@Marcin99b lub @gagyn co o tym myślicie?

Reply:
no niby spoko, tylko że tutaj za każdą iteracją tworzysz nowy obiekt List<> efektywnie zaśmiecając pamięć.

Reply:
poza tym to przyspieszy tylko w 25% przypadkach, gdyż 3 razy tworzysz listę z jednym obiektem po czym jest ona niszczona.

Reply:
weź też pod uwage to że ten kod odpala sie tylko podczas inicjalizacji a nie per wiadomość 
a to czy inicjalizacja będzie działać ułamek sekundy dłużej/krócej nie ma za bardzo znaczenia 

ale w takim przypadku gdzie zwracasz coś, co ma być tylko do odczytu
lepiej robić to jako tablice

Reply:
czyli ma być "new string[] {....}"?

Reply:
tzn. to się zwraca ostatecznie przez .ToArray() a lista (którą zmienię na HashSet) jest po to, aby dodać w drodze iteracji do niej wszystkie pola.